peps/pep-0740.rst

```diff
@@ -205,42 +180,170 @@ changes to it:
 * In addition to the current top-level ``content`` and ``gpg_signature`` fields,
   the index **SHALL** accept ``attestations`` as an additional multipart form
   field.
-* The new ``attestations`` field **SHALL** be a JSON object.
-* The JSON object **SHALL** have one or more keys, each identifying an
-  attestation format known to the index. If any key does not identify an
-  attestation format known to the index, the index **MUST** reject the upload.
-* The value associated with each well-known key **SHALL** be a JSON object.
-* Each attestation value **MUST** be verifiable by the index. If the index fails
+* The new ``attestations`` field **SHALL** be a JSON array.
+* The ``attestations`` array **SHALL** have one or more items, each a JSON object
+  representing an individual attestation.
+* Each attestation object **MUST** be verifiable by the index. If the index fails
   to verify any attestation in ``attestations``, it **MUST** reject the upload.
+  The format of attestation objects is defined under :ref:`attestation-object`
+  and the process for verifying attestations is defined under
+  :ref:`attestation-verification`.
+
+Index changes
+-------------
 
-In addition to the above, the index **SHALL** enforce a consistency
-policy for release attestations via the following:
+Simple Index
+^^^^^^^^^^^^
 
-* If the first file under a new release is supplied with ``attestations``,
-  then all subsequently uploaded files under the same release **MUST** also
-  have ``attestations``. Conversely, if the first file under a new release
-  does not have any ``attestations``, then all subsequent uploads under the
-  same release **MUST NOT** have ``attestations``.
-* All files under the same release **MUST** have the same set of well-known
-  attestation format keys.
+* When an uploaded file has one or more attestations, the index **MAY** include a
+  ``data-provenance`` attribute on its file link.
 
-The index **MUST** reject any file upload that does not satisfy these
-consistency properties.
+* When the ``data-provenance`` attribute is present, its value **SHALL** be
+  a base64 encoded :ref:`provenance object <provenance-object>`.
 
-Index changes
--------------
+JSON-based Simple API
+^^^^^^^^^^^^^^^^^^^^^
+
+* When an uploaded file has one or more attestations, the index **MAY** include a
+  ``provenance`` object in the ``file`` dictionary for that file.
+* ``provenance``, when present, **MUST** be a :ref:`provenance object <provenance-object>`.
```

Comment:
Might want to make it more clear here that only a single attestation can be considered the provenance object.

Reply:
I intended the opposite here 😅 -- a provenance object is meant to be `(identity, attestations)`, i.e. the Trusted Publisher identity that uploaded the file and one or more attestations that came from that identity (or others, such as a 3p attestation or countersignature).

Reply:
I think this could be made more clear then, this reads as if the index is exposing a single `provenance` object.

Reply:
Done -- I've emphasized that a provenance object is composed of attestations, and that the each file exposes its own provenance object.